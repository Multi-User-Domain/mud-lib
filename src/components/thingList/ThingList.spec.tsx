import React from "react";
import { render } from "@testing-library/react";
import {
  Thing,
  createThing,
  getStringNoLocale,
  addStringNoLocale,
} from "@inrupt/solid-client";

import { ThingList, IRowComponent } from "./ThingList";

const mockName = "Test Name";
const mockProperty = `http://xmlns.com/foaf/0.1/name`;
let things = [
  addStringNoLocale(createThing(), mockProperty, mockName),
  addStringNoLocale(createThing(), mockProperty, mockName),
];

it("renders", () => {
  let selectThing: (thing: Thing) => void = (thing) => {
    return null;
  };

  let MockRowComponent = function Character({
    thing,
    selectHandler,
  }: IRowComponent): React.ReactElement {
    return <p>{getStringNoLocale(thing, mockProperty)}</p>;
  };

  const { asFragment, queryAllByText } = render(
    <ThingList
      things={things}
      selectThing={selectThing}
      rowComponent={MockRowComponent}
    />
  );

  expect(queryAllByText(mockName).length).toBe(2);
  expect(asFragment()).toMatchSnapshot();
});
