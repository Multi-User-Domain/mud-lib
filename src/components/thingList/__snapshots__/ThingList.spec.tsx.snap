// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-hxjtip"
  >
    <div
      class="css-0"
    >
      <p>
        Test Name
      </p>
    </div>
    <div
      class="css-0"
    >
      <p>
        Test Name
      </p>
    </div>
  </div>
</DocumentFragment>
`;
